### PR TITLE
Remove unnecessary code&&adjust indentation

### PR DESCRIPTION
**Pull request by zhongwuzw**

### CODE REVIEW

File: Source/CarthageKit/Git.swift [hunk 1]
```diff
@@ -155,26 +155,25 @@ public func checkoutRepositoryToDirectory(
...
-		.flatMap(.concat) { environment in
-			return launchGitTask([ "checkout", "--quiet", "--force", revision ], repositoryFileURL: repositoryFileURL, environment: environment)
-		}
-		.then(SignalProducer<(), CarthageError>.empty)
+	return SignalProducer { () -> Result<[String: String], CarthageError> in
+		var environment = ProcessInfo.processInfo.environment
+		environment["GIT_WORK_TREE"] = workingDirectoryURL.path
+		return .success(environment)
+	}
+	.attempt { _ in
```
Comment (mdiep) [inline Source/CarthageKit/Git.swift]: This indentation was actually correct before. We like all the operators (like `.attempt`) to be indented one stop past the line they begin on.

File: Source/CarthageKit/Git.swift [hunk 2]
```diff
@@ -215,15 +214,14 @@ public func cloneSubmoduleInWorkingDirectory(_ submodule: Submodule, _ workingDi
...
+		repositoryCheck("remove submodule checkout") {
+			try FileManager.default.removeItem(at: submoduleDirectoryURL)
 		}
-		.then(cloneRepository(submodule.url, workingDirectoryURL.appendingPathComponent(submodule.path), isBare: false))
-		.then(checkoutSubmodule(submodule, submoduleDirectoryURL))
-		.then(purgeGitDirectories)
+	}
+	.then(cloneRepository(submodule.url, workingDirectoryURL.appendingPathComponent(submodule.path), isBare: false))
+	.then(checkoutSubmodule(submodule, submoduleDirectoryURL))
+	.then(purgeGitDirectories)
```
Comment (mdiep) [inline Source/CarthageKit/Git.swift]: Same here.

File: Source/CarthageKit/Project.swift [hunk 3]
```diff
@@ -1213,57 +1213,56 @@ public func cloneOrFetch(
...
-									fetchRepository(repositoryURL, remoteURL: remoteURL, refspec: "+refs/heads/*:refs/heads/*")
-										.then(SignalProducer<(ProjectEvent?, URL), CarthageError>.empty)
-								)
+	return SignalProducer {
+		Result(at: destinationURL, attempt: {
+			try fileManager.createDirectory(at: $0, withIntermediateDirectories: true)
+			return dependency.gitURL(preferHTTPS: preferHTTPS)!
+		})
+	}
+	.flatMap(.merge) { (remoteURL: GitURL) -> SignalProducer<(ProjectEvent?, URL), CarthageError> in
```
Comment (mdiep) [inline Source/CarthageKit/Project.swift]: And here too.

File: Source/CarthageKit/Resolver.swift [hunk 4]
```diff
@@ -188,32 +188,30 @@ public struct Resolver: ResolverProtocol {
...
-					.observe(on: scheduler)
-					.permute()
-					.flatMap(.concat) { graphs -> SignalProducer<Signal<DependencyGraph, CarthageError>.Event, NoError> in
-						return SignalProducer<DependencyGraph, CarthageError>
-							{
-								mergeGraphs([ inputGraph ] + graphs)
-							}
-							.materialize()
+		}
+		.flatMap(.concat) { graph, nodes -> SignalProducer<DependencyGraph, CarthageError> in
```
Comment (mdiep) [inline Source/CarthageKit/Resolver.swift]: And here.

Comment (zhongwuzw) [reply Source/CarthageKit/Resolver.swift]: @mdiep , I updated the indentation.

**Comment by mdiep** [comment]
Thanks again for the PR!

**Comment by zhongwuzw** [comment]
@mdiep :beers::beers:

**Pull request merged**